reconciler/reconciler.go

```diff
@@ -101,27 +102,29 @@ const (
 // what sort of storage layer they want to use to provide the required
 // information.
 type Helper interface {
+	CurrentBlock(
+		ctx context.Context,
+	) (*types.BlockIdentifier, storage.DatabaseTransaction, error)
```

Comment:
I don't have full knowledge of how this fits together, but I generally would prefer that whatever is creating the db transaction is also responsible for the commit or rollback. Having one function create a tx and then return it seems a little dangerous since it no longer has any control over the tx. Does the reconciler know enough about `storage` to be able to create the transaction itself and pass it around?

Reply:
> I don't have full knowledge of how this fits together, but I generally would prefer that whatever is creating the db transaction is also responsible for the commit or rollback.

The storage layer can't handle the commit/rollback because it doesn't know when we are done using this tx (it spans multiple calls). The alternative is to have the reconciler create the transaction and pass it to `CurrentBlock` instead of getting it back.

> Does the reconciler know enough about storage to be able to create the transaction itself and pass it around?

This is a pretty common for packages in `rosetta-sdk-go` for dealing with situations that require database consistency. (I'm not sure there is another way to do these ops without having the packages control the db tx). Take a look at this file for a more complex example of this (it has a separate helper for `DatabaseTransaction` creation): https://github.com/coinbase/rosetta-sdk-go/blob/master/constructor/coordinator/coordinator.go